lib/navigation/crosscheck.hpp

```diff
@@ -0,0 +1,22 @@
+#include <array>
+
+namespace hyped::navigation{
+
+  class Crosscheck{
```

Comment:
formatting

Reply:
also, what's the purpose of this? maybe add a doc comment? in particular, comment in the header what the class is supposed to achieve and comment on the functions in the source file how it is achieve, if that's not obvious from reading the code (example of obvious method: linear regression; example of non-obvious method: Kalman filter)

Reply:
Purpose of this is to check the sensor readings against each other (i.e. does the double integrated value of acceleration agree with the value form the wheel encoders). This will be called in the main navigation function between the preprocessing step and updating the current trajectory. Ngl the name might be a bit misleading but couldn't think of anything better so open to suggestions.

Reply:
Yeah not really much point in it if all we do is pass in the integrated estimates from each sensor and just make a function to compare them, surely this could be done in about 3 lines in the navigator class without having to push a load of data around? Still working on how much/ how/ where to store historic data but it would probably be in here and we would use this class to call integrations/ complete all conversions and then compare.

Reply:
The point of this is to keep the messy background work that is needed to compare the sensor's estimates away from the neat navigator class so that all we need to do is call a function from here, pass over the current data and let it do it's job. Idk if there's a nicer way of doing things tho?

Reply:
Sounds like it would require some experimentation. I will say though: A class should have exactly one job. If that job is doing numerical calculations (integration, differentiation, ...), storing data, and comparing it then something has gone wrong.

Classes aren't expensive so feel free to make more of them. If some of them turn out to be useless we'll clean them up.

Reply:
The integration itself would probably get put in utils or something and then just used within this class. The singular purpose of this class is to check that we still know what's going on. Will tidy up data a bit for now and then leave this until the point of implementation since we know what it should do and can volt on anything extra we need at that point in time?